### PR TITLE
[Feature][RSC-Bots][1745625]

### PR DESCRIPTION
**Pull request by v-rodriguezd**
Created document for Bot APIs for RSC permissions. Please review.

### CODE REVIEW

File: msteams-platform/TOC.yml [hunk 1]
```diff
@@ -213,6 +213,9 @@
       href: bots/how-to/conversations/conversation-basics.md
     - name: Messages in bot conversations
       href: bots/how-to/conversations/conversation-messages.md
+      items:
+      - name: Receive all messages with RSC
```
Comment (vule-msft) [inline msteams-platform/TOC.yml:217]: Suggest updating text to - 
"Receive all channel messages with RSC"

File: msteams-platform/TOC.yml [hunk 2]
```diff
@@ -213,6 +213,9 @@
...
+        href: bots/how-to/conversations/bot-apis-with-rsc.md
```
Comment (vule-msft) [inline msteams-platform/TOC.yml:218]: This is already under the "bots/how-to/conversations" path. The doc is specific to channel messages with RSC. 

Suggest to update the path to something more content specific - "channel-messages-with-rsc"

File: msteams-platform/TOC.yml [hunk 3]
```diff
@@ -363,6 +366,8 @@
     href: https://docs.microsoft.com/graph/toolkit/get-started/build-a-microsoft-teams-tab
   - name: Enable resource-specific consent
     href: graph-api/rsc/resource-specific-consent.md
+  - name: Bot APIs with RSC permissions
```
Comment (vule-msft) [inline msteams-platform/TOC.yml:369]: "Bots with RSC permissions"

File: msteams-platform/bots/how-to/conversations/bot-apis-with-rsc.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+---
+
+# Receive all channel messages with RSC
+
+> [!NOTE]
+> This feature is available in public developer preview.
+
+The resource-specific consent (RSC) permissions model, originally developed for Teams Graph APIs, is now extended to bot scenarios.
+
+Currently, bots can only receive user channel messages when they are @mentioned. Using RSC, you can now request team owners to consent for a bot to receive user messages across all channels in a team without using @mention. This capability is enabled by specifying the `ChannelMessage.Read.Group` permission in the manifest of an RSC enabled Teams app. After configuration, Team owners can grant consent during the app installation process.
```
Comment (vule-msft) [inline msteams-platform/bots/how-to/conversations/bot-apis-with-rsc.md:16]: > receive user messages across all channels

Please specify "standard channels". This feature does not apply to private channels

> in a team without using @mention

in a team without ~~using~~ being @mentioned

File: msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md [hunk 5]
```diff
@@ -31,7 +31,10 @@ Next, you can retrieve mentions using the `entities` object and add mentions to
 
 ## Work with mentions
 
-Every message to your bot from a group or channel contains an @mention with its name in the message text. Ensure that your message parsing handles @mention. Your bot can also retrieve other users mentioned in a message and add mentions to any messages it sends.
+> [!NOTE]
```
Comment (vule-msft) [inline msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md:34]: We can remove the NOTE dialog here. It's already called out in the first section, above

File: msteams-platform/TOC.yml [hunk 6]
```diff
@@ -363,6 +366,8 @@
...
+    href: bots/how-to/conversations/bot-apis-with-rsc.md
```
Comment (vule-msft) [inline msteams-platform/TOC.yml:370]: update with new link

File: msteams-platform/whats-new.md [hunk 7]
```diff
@@ -149,6 +149,7 @@ Developer preview is a public program that provides early access to unreleased T
 
 | **Date** | **Notes** | **Changed topics** |
 | -------- | --------- | ------------------ |
+|05/11/2021|Bots can be enabled to receive all channel messages using resource-specific consent (RSC).|[Receive all messages with RSC](~/bots/how-to/conversations/bot-apis-with-rsc.md), [bot conversation overview](~/bots/how-to/conversations/conversation-basics.md), [channel and group conversations](~/bots/how-to/conversations/channel-and-group-conversations.md), and [developer preview manifest schema](~/resources/schema/manifest-schema-dev-preview.md) |
```
Comment (vule-msft) [inline msteams-platform/whats-new.md:152]: please double check links

> bot-apis-with-rsc.md

should be updated

**Review by vule-msft — CHANGES_REQUESTED** [review]

File: msteams-platform/TOC.yml (hunk 3 above)
Comment (v-rodriguezd) [reply msteams-platform/TOC.yml:369]: This is an error from my end, I should have removed this entry as it is duplicated in two places. As recommended by you keeping it under bots/how-to/conversations/channel-messages-with-rsc.md. Removing this duplicate entry.

File: msteams-platform/TOC.yml (hunk 6 above)
Comment (v-rodriguezd) [reply msteams-platform/TOC.yml:370]: Removed this entry.

File: msteams-platform/bots/how-to/conversations/channel-messages-with-rsc.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+"webApplicationInfo": {
+"id": "XXxxXXXXX-XxXX-xXXX-XXxx-XXXXXXXxxxXX",
+"resource": "https://AnyString",
+"applicationPermissions": [
+"ChannelMessage.Read.Group"
+    ]
+  }
+```
+
+The following image illustrates how to create the zip file for the app manifest:
```
Comment (vule-msft) [inline msteams-platform/bots/how-to/conversations/channel-messages-with-rsc.md:48]: Let's remove this along with the image below it. Developers should already be familiar with how to create a zip from the manifest

File: msteams-platform/bots/how-to/conversations/channel-messages-with-rsc.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+![Create zip file](~/bots/how-to/conversations/Media/createzipfile.png)
+
+## Receive channel messages with RSC
```
Comment (vule-msft) [inline msteams-platform/bots/how-to/conversations/channel-messages-with-rsc.md:52]: Would it make sense for this to be updated to "Side-load in team to test"?

**Review by vule-msft — CHANGES_REQUESTED** [review]

**Review by vule-msft — APPROVED** [review]

**Review by v-abirade — APPROVED** [review]
Approved

**Review by surbhigupta12 — APPROVED** [review]
Approved.

**Pull request merged**